### DISCUSSION TITLE
What could be improved with the command line utilities?

### DISCUSSION THREAD

**Discussion by dsommers** [OpenVPN/openvpn3-linux#159] [General]
We have now just released OpenVPN 3 Linux v19_beta and are preparing the codebase for a coming stable release.

Now the project is in the phase where it starts to explore further changes to improve the user experience.  These changes will be considered for the development after the first stable release is out.

The bigger channges on our wish list are:

* NetworkManager OpenVPN 3 plug-in
* [Cockpit](https://cockpit-project.org/) OpenVPN 3 package
* Proper PKCS#11 support
* Proper PKCS#12 support (requires [OpenVPN 3 Core](https://github.com/OpenVPN/openvpn3) library changes)

What would you like to see?  What "hacks" to you use for your daily OpenVPN 3 Linux usage?

**Comment by Christiaan676** [comment]
Support for Keyring services would be nice.

**Reply by dsommers** [reply]
Agreed!  And this was [discussed a bit back](https://github.com/OpenVPN/openvpn3-linux/pull/114), but unfortunately it seems to have stranded for some reason.
